### PR TITLE
Fix missing supported operations #2689

### PR DESCRIPTION
**Pull request by albinsuresh**
## Proposed changes

The bug was caused by sending messages to `c8y/s/us` topic before the c8y-bridge connection is even established, resulting in those messages getting lost. With this fix we make sure that the mapper does not start mapping or even send any init messages to the cloud until the bridge connection is established.

- [x] System test
- [x] Fix

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue

https://github.com/thin-edge/thin-edge.io/issues/2689

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: tests/RobotFramework/tasks/E2E_Tutorials/Connect_to_c8y.robot [hunk 1]
```diff
@@ -39,7 +39,6 @@ Create the certificate
     Should Contain    ${output}    Thumbprint:
 
 tedge cert upload c8y command
-    Execute Command    sudo tedge cert create --device-id ${DEVICE_SN}
```
Comment (albinsuresh) [inline tests/RobotFramework/tasks/E2E_Tutorials/Connect_to_c8y.robot:42]: Already done in the previous step.

File: crates/core/tedge_mapper/src/c8y/mapper.rs [hunk 2]
```diff
@@ -51,6 +51,13 @@ impl TEdgeComponent for CumulocityMapper {
         let mut uploader_actor = UploaderActor::new(identity.clone()).builder();
         let mut downloader_actor = DownloaderActor::new(identity).builder();
 
+        // MQTT client dedicated to monitor the c8y-bridge client status and also
+        // set service down status on shutdown, using a last-will message.
+        // A separate MQTT actor/client is required as the last will message of the main MQTT actor
+        // is used to send down status to health topic.
+        let mut service_monitor_actor =
+            MqttActorBuilder::new(service_monitor_client_config(&tedge_config)?);
```
Comment (albinsuresh) [inline crates/core/tedge_mapper/src/c8y/mapper.rs:59]: This MQTT actor which was previously used just to send the last will message is reused to monitor the bridge health status as well.

File: crates/core/c8y_api/src/utils.rs
Comment (Bravo555) [inline crates/core/c8y_api/src/utils.rs:1]: note: IMO `c8y_api` crate is a bad place for bridge related utilities as name suggests the crate handles cumulocity-specific stuff, e.g. authentication, smartrest, inventory, etc.

This in itself is a low priority, but more important is that checking if the bridge is established should also work for AWS and Azure bridges, but as this change only handles cumulocity, does it mean that this issue can also appear for Azure or AWS? If so, then we should consider fixing it there as well as a follow-up.

File: tests/images/debian-systemd/files/bootstrap.sh [hunk 3]
```diff
@@ -14,6 +14,7 @@ FLAGS
     WORKFLOW FLAGS
     --clean/--no-clean                      Clean the device of any existing tedge installations before installing/connecting. Default False
     --install/--no-install                  Install thin-edge.io. Default True
+    --bootstrap/--no-bootstrap              Bootstrap tedge by setting c8y.url and creating and uploading the device certificate. Default True
```
Comment (Bravo555) [inline tests/images/debian-systemd/files/bootstrap.sh:17]: todo: delete line 21 which also contains `--bootstrap`

**Review by Bravo555 — CHANGES_REQUESTED** [review]
One small change required in `bootstrap.sh`

File: crates/core/c8y_api/src/utils.rs
Comment (albinsuresh) [reply crates/core/c8y_api/src/utils.rs:1]: > note: IMO c8y_api crate is a bad place for bridge related utilities as name suggests the crate handles cumulocity-specific stuff, e.g. authentication, smartrest, inventory, etc.

I agree. I think it was added in this crate for lack of another common crate that could be accessed by all the c8y components like the mapper and other c8y plugins. But, now that the other components are gone, I don't see anything wrong with moving this into the mapper. But, that being said, this bridge itself is going to get replaced with our own bridge component in the near future. So, may not be a high priority item for now.

> This in itself is a low priority, but more important is that checking if the bridge is established should also work for AWS and Azure bridges, but as this change only handles cumulocity, does it mean that this issue can also appear for Azure or AWS? If so, then we should consider fixing it there as well as a follow-up.

The principle definitely applies to other mappers as well, that no mapping should happen until the bridge is established. But, this is less critical in the case of other mappers, in their current state, as they only map telemetry data, and not entity registrations or any other critical metadata messages. It's okay to drop a few telemetry messages before the bridge is established, as updates to them would be coming later on. But, that's not the case with other entity metadata messages which are only sent once and that's why it's far more critical for the c8y mapper.

**Review by Bravo555 — APPROVED** [review]

**Pull request merged**